inception-search-mtas/src/main/java/de/tudarmstadt/ukp/inception/search/index/mtas/MtasUimaParser.java

```diff
@@ -242,68 +242,158 @@ else if (annotation instanceof Sentence) {
+                                    MtasToken mtasAnnotationTypeFeatureLabel = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationTypeFeatureLabel.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationTypeFeatureLabel.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationTypeFeatureLabel);
+
+                                    indexedStr = annotationUiName.replace(" ", "_") + "."
+                                            + feature.getUiName().replace(" ", "_")
+                                            + MtasToken.DELIMITER + kbValue;
+
+                                    // Indexing UI annotation without type i.e Concept/Instance
+                                    log.debug("Indexed String without type for : {}", indexedStr);
+                                    MtasToken mtasAnnotationFeatureLabel = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationFeatureLabel.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationFeatureLabel.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationFeatureLabel);
+                                    
+                                    // Indexing UI annotation without type and layer for generic search
+                                    indexedStr = IndexingConstants.kbEntity + MtasToken.DELIMITER
+                                            + kbValue;
+                                    log.debug("Indexed String without type and label for : {}", indexedStr);
+                                    MtasToken mtasAnnotationKBEntity = new MtasTokenString(
+                                            mtasId++, indexedStr, beginToken);
+                                    mtasAnnotationKBEntity.setOffset(annotation.getBegin(),
+                                            annotation.getEnd());
+                                    mtasAnnotationKBEntity.addPositionRange(beginToken,
+                                            endToken);
+                                    tokenCollection.add(mtasAnnotationKBEntity);
+                                    
+                                }
+
                             }
                         }
                     }
                 }
             }
-        }
-        catch (Exception e) {
+        } catch (Exception e) {
             log.error("Unable to index document", e);
```

Comment:
Provide additional information about the document. Cf.

```
                log.info(
                        "Created annotation document [{}] for user [{}] for source document "
                        + "[{}]({}) in project [{}]({})",
                        aAnnotationDocument.getId(), aAnnotationDocument.getUser(), 
                        aAnnotationDocument.getDocument().getName(),
                        aAnnotationDocument.getDocument().getId(),
                        aAnnotationDocument.getProject().getName(),
                        aAnnotationDocument.getProject().getId());``

Reply:
Try block is the outer block. Will figure out when refracting the code.